pkg/tnf/reel/reel.go

```diff
@@ -206,32 +205,31 @@ func (r *Reel) Step(step *Step, handler Handler) error {
 		} else {
 			if len(results) > 0 {
 				result := results[0]
-
 				output, outputStatus := r.stripEmulatedPromptFromOutput(result.Output)
 				if outputStatus != 0 {
 					return fmt.Errorf("error executing command exit code:%d", outputStatus)
 				}
 				match, matchStatus := r.stripEmulatedPromptFromOutput(result.Match[0])
 				log.Debugf("command status: output=%s, match=%s, outputStatus=%d, matchStatus=%d", output, match, outputStatus, matchStatus)
 
-				matchIndex := strings.Index(output, match)
-				var before string
-				// special case:  the match regex may be nothing at all.
-				if matchIndex > 0 {
-					before = output[0 : matchIndex-1]
+				if result.CaseIdx != len(batcher[0].Cases())-1 {
```

Comment:
batcher[0] has the Args() BSnd batcher, which is used to send the handler's Args() command. You should use batcher[len(batcher) - 1] or similar, to avoid handler's that don't have an Args() command.